Lesson2/src/Homework2.java

```diff
@@ -20,13 +20,16 @@ public static void main(String[] args) {
      **/
     public static int sum(int a, int b) {
         int x = a + b;
-        int z = 0;
-        long y = Integer.MAX_VALUE + Integer.MAX_VALUE;
-        z = (int) y;
-        if (x > z)
-            return x;
-        else
+
+        int i = Integer.MAX_VALUE;
+        long l = (long) (i);
+        long y = i + l;
```

Comment:
check this.

long a = Integer.MAX_VALUE + 1; //value of a = Integer.MIN_VALUE, because  Integer.MAX_VALUE + 1 will be casted to int
long b = Integer.MAX_VALUE + (long)1; value of b =  Integer.MAX_VALUE + 1 because operation with at least 1 long will be casted to long

so you need to do smth like 

long y = (long)Integer.MAX_VALUE + Integer.MAX_VALUE; 
to get long variable with value of )Integer.MAX_VALUE + Integer.MAX_VALUE